File: C++Programming/Practice/program47.cpp
// Accept number from user and calculate sum of its factors

#include<iostream>
using namespace std;

class NumberX
{
    public :
        int SumFactors(int iNo)
        {
            int iCnt = 0;
            int iSum = 0;

            for(iCnt = 1; iCnt <= (iNo / 2); iCnt++)
            {
                if(iNo % iCnt == 0)
                {
                    iSum = iSum + iCnt;
                }
            }

            return iSum;

        }
};

int main()
{
    int iValue = 0;
    int iRet = 0;

    cout<<"Enter number : ";
    cin>>iValue;

    NumberX nobj;

    iRet = nobj.SumFactors(iValue);

    cout<<"Sum of factors is : "<<iRet<<endl;

    return 0;
}

// Time complexity : O(n/2)
// where n >= 0

File: Java_Programming/Practice/program47.java
// Accept number from user and display its factors

import java.util.Scanner;

class NumberX
{
    public void DisplayFactors(int iNo)
    {
        int iCnt = 0;

        for(iCnt = 1; iCnt < iNo; iCnt++)
        {
            if(iNo % iCnt == 0)
            {
                System.out.println(iCnt);
            }
        }


    }

}
class program47
{
    public static void main(String A[])
    {
        int iValue = 0;

        Scanner sobj = new Scanner(System.in);

        System.out.println("Enter number : ");
        iValue = sobj.nextInt();

        NumberX nobj = new NumberX();

        nobj.DisplayFactors(iValue);

        

    }
}

// Time complexxity : O(n)
// where n >= 0
